src/wasm-features.h

```diff
@@ -134,7 +139,16 @@ struct FeatureSet {
   void setTypedFunctionReferences(bool v = true) {
     set(TypedFunctionReferences, v);
   }
-  void setAll(bool v = true) { features = v ? All : MVP; }
+  void setGCNNLocals(bool v = true) { set(GCNNLocals, v); }
+  void setAll(bool v = true) {
+    features = v ? All : MVP;
+    // By default disable GCNNLocals, forcing the user to opt in to that feature
+    // explicitly. That is, wasm-opt -all will enable GC but *not* enable
+    // non-nullable locals. To get them, do wasm-opt -all --enable-gc-nn-locals
+    // FIXME: When the wasm spec stabilizies, this feature will go away, and
+    //        this awkwardness with it.
+    setGCNNLocals(false);
```

Comment:
It might be simpler to just have `All` not include the bit for `GCNNLocals`. Then whether `GCNNLocals` is enabled would be completely independent of all other feature flags, which I think is a simpler mental model.

Reply:
I considered that, but it is riskier and would require more work, because there are a few places that loop over `All` (for example, printing), and those would break.

Instead I think it is pretty clear mentally to only modify `setAll`. That is, this new feature is a normal feature, except for the one oddity of not being set by default in `setAll`.

Reply:
What if instead of unconditionally unsetting the feature in `setAll`, we instead just left it unchanged? That would be the simpler mental model IMO without the implementation downsides you mentioned.

Reply:
Does `--enable-gc-nn-locals -all` actually disable gc-nn-locals then?

Reply:
@tlively 

> What if instead of unconditionally unsetting the feature in setAll, we instead just left it unchanged?

The downside there is that then `wasm-opt -all` would enable non-nullable locals. A specific consequence of that is that we would not run the code to add in `ref.as_non_null`s to fix up new locals that we add that are non-nullable (as many passes do). And then our output would not work in VMs by default. Basically all our current users would need to either start using the non-nullable experiment in v8 as well, or start passing `wasm-opt -all --disable-wasm-nn-locals`.

As this is an experiment, I think it makes sense instead to not break out existing users, and to make this opt-in.

@aheejin 

> Does --enable-gc-nn-locals -all actually disable gc-nn-locals then?

Oh - good point! They would run in order, and so we would actually disable it... I'll fix that.

Reply:
> The downside there is that then wasm-opt -all would enable non-nullable locals.

I would say the root issue causing that is that the default feature set includes non-nullable locals.

Reply:
Fair enough, but then we return to the previous point that if we remove it from `All` then we'd need to update code in every current place that loops over All, like printing of features, iteration on features, etc. That's a much more intrusive change.

If we had a long-term use for this, then I'd suggest a larger refactor like adding a new set of ExperimentalFeatures that is handled differently than the normal ones. But as this is pretty temporary, and this PR works properly, and it seems to have little risk, I'd prefer to keep it as is. If you feel strongly though I can look into that refactoring. And e.g. if we think we'll have new experimental features over time then it might pay off eventually.